udata/core/dataset/models.py

```diff
@@ -477,6 +477,8 @@ class Dataset(WithMetrics, BadgeMixin, db.Owned, db.Document):
 
     featured = db.BooleanField(required=True, default=False)
 
+    contact_point = db.ReferenceField('ContactPoint', reverse_delete_rule=db.PULL)
```

Comment:
I had this issue occurring when deleting a contact point that had been attached to a dataset:
```
Update failed (Cannot apply $pull to a non-array value, full error: {'index': 0, 'code': 2, 'errmsg': 'Cannot apply $pull to a non-array value'})
```